core/models.py

```diff
@@ -4,3 +4,43 @@
 
 class User(AbstractUser):
     pass
+
+
+class Genre(models.Model):
+    name = models.CharField(max_length=26)
+    is_for_adults = models.BooleanField(default=False)
+
+
+class Actor(models.Model):
+    name = models.CharField(max_length=60)
+    age = models.PositiveSmallIntegerField(help_text="Age of the actor in years")
+    nationality = models.CharField(max_length=54)
+
+
+class Movie(models.Model):
+    title = models.CharField(max_length=185)
+    genres = models.ManyToManyField(Genre, related_name="movies")
+    actors = models.ManyToManyField(Actor, related_name="movies")
+    duration = models.PositiveIntegerField(help_text="Duration of the movie in minutes")
+
+
+class CinemaHall(models.Model):
+    name = models.CharField(max_length=20)
+
+
+class Seat(models.Model):
+    row = models.PositiveSmallIntegerField()
+    number = models.PositiveSmallIntegerField()
+    hall = models.ForeignKey(CinemaHall, on_delete=models.PROTECT)
```

Comment:
pls add name for all related fields, not only m2m. It is handy, because django's default naming is questionable...